src/style/calc_dimension.rs

```diff
@@ -0,0 +1,58 @@
+//! Dimensions that are have to be calculated while resolving.
+
+use crate::resolve::MaybeResolve;
+
+use super::Dimension;
+
+/// An addition, subtraction, multiplication or division between [`Dimension`]s.
+///
+/// The values are calulated when the point values are resolved.
+///
+/// This type is needed, because not all calculations can be determined before resolving the actual values.
+/// For example, when adding [`Dimension::Points`] and [`Dimension::Percent`] together,
+/// the percentage first needs to be resolved to an absolute value to get the final result.
+#[derive(Clone, PartialEq, Debug)]
+pub enum CalcDimension {
+    /// Add two [`Dimension`]s together.
+    Add(Box<Dimension>, Box<Dimension>),
```

Comment:
Hmm, maybe. It will still work for `Undefined` and `Auto`, but the result will always be the same. I'd like to avoid duplicating some of the enum values, but I guess the current approach is also not ideal.